wiki-migration, new section to cli page (#3422)
jenkins-infra/jenkins.io: content/doc/book/managing/cli.adoc

%
----

== Using the CLI client

While the SSH-based CLI is fast and covers most needs, there may be situations where the CLI
client distributed with Jenkins is a better fit. For example, the default transport for the CLI client
is HTTP which means no additional ports need to be opened in a firewall for its
use.

> ==== Comparing SSH and CLI client

Reviewer: Heading levels are expected increase and decrease by only one.
```suggestion
=== Comparing SSH and CLI client
```

Both SSH and jenkins-cli.jar provide access to a set of commands that lets you interact with Jenkins from a command line, but they have a few differences:

* Jenkins SSH does not require any custom jar file on the client side, making it easier to access Jenkins from a variety of sources

> * Jenkins CLI client can be intelligent, performing file access and other interactive processing, whereas SSH client is dumb. Therefore, some commands can only run through Jenkins CLI and not via SSH

Reviewer 2: My concern here is with the verbiage. I am not a fan of terms such as `dumb`. I would prefer this read something like `whereas the SSH client is not as intuitive`
Thoughts?

Author: humm. I guess it is a valid suggestion. What about: 

"SSH client was build to be a generic tool to serve several propouses, so it don't offer no easy-way to do basic things that are common and specific to jenkins enviroment. Therefore use Jenkins CLI maybe will increase your productivity and your user experience."

I'm not a english native speaker, so please help me if you find any sintaxe or semantic error. ;)

Reviewer 2: I like that very much. Thank you 🙂

Author: Did it. Thank you.

> * SSH client was build to be a generic tool to serve several propouses, so it don't offer an easy-way to do basic things that are common and specific to jenkins enviroment. Therefore use Jenkins CLI maybe will increase your productivity and your user experience.

Reviewer: ```suggestion
* SSH client was build to be a generic tool to serve several purposes.
It doesn't offer an easy way to do basic things that are common and specific to Jenkins environments
Using the `jenkins-cli.jar` instead of the ssh client may increase productivity and improve the development experience
```

%
----

== Using the CLI client

While the SSH-based CLI is fast and covers most needs, there may be situations where the CLI
client distributed with Jenkins is a better fit. For example, the default transport for the CLI client
is HTTP which means no additional ports need to be opened in a firewall for its
use.

=== Comparing SSH and CLI client

Both SSH and jenkins-cli.jar provide access to a set of commands that lets you interact with Jenkins from a command line, but they have a few differences:

> * Jenkins SSH does not require any custom jar file on the client side, making it easier to access Jenkins from a variety of sources
> * SSH client was build to be a generic tool to serve several purposes.
> It doesn't offer an easy way to do basic things that are common and specific to Jenkins environments
> Using the `jenkins-cli.jar` instead of the ssh client may increase productivity and improve the development experience
> 

Reviewer 3: ```suggestion
* Jenkins SSH does not require any custom jar file on the client side, making it easier to access Jenkins from a variety of sources.
* SSH client was build to be a generic tool to serve several purposes.
It doesn't offer an easy way to do basic things that are common and specific to Jenkins environments.
Using the `jenkins-cli.jar` instead of the ssh client may increase productivity and improve the development experience.

```
